services/tdei.ts

```diff
@@ -209,6 +209,13 @@ export class TdeiClient extends BaseHttpClient implements ICancelableClient {
     return (await response.json())[0];
   }
 
+  async getDatasetsByProjectGroupAndName(projectGroupId: string, name: string) {
+    const response = await this._get(`datasets?tdei_project_group_id=${projectGroupId}&name=${encodeURIComponent(name)}`);
+
+    return (await response.json())
+      .map(d => ({ id: d.tdei_dataset_id, name: d.metadata.dataset_detail.name }));
+  }
```

Comment:
Maybe a future optimization: since we received all of the dataset metadata here already, we could avoid the extra requests/latency on the "create" page to re-fetch these for each selection. For now, your approach that minimizes changes to the existing implementation should be fine.